packages/core/vue/src/useWallet.ts

```diff
@@ -0,0 +1,225 @@
+import { ref, Ref, computed, watch, watchEffect } from '@vue/runtime-core';
+import { Connection, PublicKey, Transaction, TransactionSignature } from '@solana/web3.js';
+import {
+    SendTransactionOptions,
+    WalletNotConnectedError,
+    WalletNotReadyError,
+    WalletError,
+} from '@solana/wallet-adapter-base';
+import { Wallet, WalletName } from '@solana/wallet-adapter-wallets';
+import { useLocalStorage } from './useLocalStorage';
+import { WalletNotSelectedError, OperationNotSupportedByWalletError } from './errors';
+
+type Adapter = ReturnType<Wallet['adapter']>;
+type WalletDictionary = { [key: string]: Wallet };
+
+export interface WalletStore {
+    // Props.
+    wallets: Wallet[];
+    autoConnect: boolean;
+
+    // Data.
+    walletName: Ref<string | null>,
+    walletsByName: Ref<WalletDictionary>,
+    wallet: Ref<Wallet | null>;
+    adapter: Ref<Adapter | null>;
+    publicKey: Ref<PublicKey | null>;
+    ready: Ref<boolean>;
+    connected: Ref<boolean>;
+    connecting: Ref<boolean>;
+    disconnecting: Ref<boolean>;
+
+    // Methods.
+    select(walletName: WalletName): void;
+    connect(): Promise<void>;
+    disconnect(): Promise<void>;
+    sendTransaction(transaction: Transaction, connection: Connection, options?: SendTransactionOptions): Promise<TransactionSignature>;
+    signTransaction(transaction: Transaction): Promise<Transaction>;
+    signAllTransactions(transaction: Transaction[]): Promise<Transaction[]>;
+    signMessage(message: Uint8Array): Promise<Uint8Array>;
+}
+
+export interface WalletStoreProps {
+    wallets: Wallet[];
+    autoConnect?: boolean;
+    onError?: (error: WalletError) => void;
+    localStorageKey?: string;
+}
+
+let walletStore: WalletStore = {} as WalletStore;
+
+export const useWallet = (): WalletStore => walletStore;
+
+export const initWallet = ({
+    wallets,
+    autoConnect = false,
+    onError = (error: WalletError) => console.error(error),
+    localStorageKey = 'walletName',
+}: WalletStoreProps): void => {
+    const walletName = useLocalStorage<string>(localStorageKey);
+    const wallet = ref<Wallet | null>(null);
+    const adapter = ref<Adapter | null>(null);
+    const publicKey = ref<PublicKey | null>(null);
+    const ready = ref<boolean>(false);
+    const connected = ref<boolean>(false);
+    const connecting = ref<boolean>(false);
+    const disconnecting = ref<boolean>(false);
```

Comment:
I'm not sure how this applies to Vue, but the way this state is handled has changed in the React package:
https://github.com/solana-labs/wallet-adapter/blob/581cb52a41ad7a255542de39115d051b0833e57c/packages/core/react/src/WalletProvider.tsx#L42

This allows all related state updates to happen at once, which prevents inconsistent state, rerenders, and race conditions. I see you are updating the state below in one place which is great.

It may be worth looking at how the `connecting`/`disconnecting`/`working` values are handled now also to avoid rerenders or `useEffect` invocations (again with the caveat that I don't know how relevant to Vue this is).

Reply:
I can do something similar but I don't think this will change anything in Vue because the code will be executed within the same DOM update cycle (a.k.a a tick).

But I can of course group it in a state object so that it is at least consistent with the logic of the other packages.

---

On a similar note, I had issues with reactivity if I didn't provide the `Refs` directly in the `WalletStore` but I'll try again to give the real values instead of the `Ref` variables.

To give more context, in Vue `Refs` are equivalent to `useState` but the API is quite different:

```js
// In React.
const [foo, setFoo] = useState('defaultFoo'); 
foo; // Access
setFoo('newFoo'); // Edit

// In Vue.
const foo = ref('defaultFoo');
foo.value; // Access
foo.value = 'newFoo'; // Edit
```

So ultimately, by returning the `Ref` we are also giving the equivalent of the `setFoo` method which I don't think is what we want. I'll see what I can do. 🙂

Reply:
Okay I've done some tests and If I don't return `Refs` reactivity simply does not work in Vue for primitive values.

An alternative would be to wrap the entire store in the `reactive` function, but that would still allow us to update any value inside it (including functions) and would differ from what most people are doing when creating Vue 3 libraries.

Also, the nice thing about returning `Refs` is that they are perfectly understood by Vue's components and their template knows how to handle them even without using `.value`.

Reply:
Ah, okay. The behavior you had doesn't necessarily need to be changed, as long as nothing can react to those individual value changes before they are all changed.